packages/nvidia_nat_vanna/src/nat/plugins/vanna/text2sql.py

```
@@ -0,0 +1,249 @@
+                    payload=StatusPayload(message="Executing SQL query...").model_dump_json(),
+                )
+                # Execute SQL and propagate errors
+                # Note: run_sql is dynamically set as async function in setup_vanna_db_connection
+                df = await vanna_instance.run_sql(sql)  # type: ignore[misc]
+                logger.info(f"SQL executed successfully: {len(df)} rows returned")
+
+            # Yield final result as Text2SQLOutput
+            yield Text2SQLOutput(sql=sql, explanation=explanation)
+
+        except Exception as e:
+            logger.error(f"SQL generation failed: {e}")
```

Ruff (0.14.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)